includes/CAPx/Drupal/Importer/EntityImporter.php

```diff
@@ -168,9 +168,9 @@ public function justDoIt() {
 
     if (!empty($data)) {
       foreach ($data as $type => $results) {
+        drupal_alter('capx_preprocess_results', $results, $this);
```

Comment:
Thanks for the fix on the typo.

Moving this line seems reasonable and the API documentation seems to support it but this is a change to how the alter function is expected to be used. In the previous position this alter only had a scope of one result item and would not have the ability to compare items against others in the response. What kind of preprocessing are you doing if you don't mind me asking.

Reply:
Thanks for the explanation @ccjjmartin,

I've got a couple of things that might make your life easier. If you haven't already, I would employ to you explore the https://github.com/SU-SWS/stanford_capx/tree/7.x-3.x/modules/capx_filters module. This module provides an interface for creating filters to prevent saving profiles to your site. You can combine this module with the https://github.com/SU-SWS/stanford_capx/tree/7.x-3.x/modules/capx_tampers module which has feeds like tampering functionality to narrow your imported result set down quite a bit. More help on those module can be found in our user guide: https://sites-jumpstart.stanford.edu/jumpstart-user-guide/stanford-content/people/filtering-imported-profiles-capx

The too many users issue is a real problem as we have quickly overrun available resources on both the API side and the client side when pulling in orgs with too many users. Until about 2 weeks ago this was the way life was. Recently the API team has release new functionality that will help with this as they have provided a way to filter out results on their end as well as provide a way to allow us to request only parts of the response data for each profile. Work on our end hasn't yet started. In the mean time, you can lower the number of profiles processed per operation by adjusting the `stanford_capx_batch_limit` variable.